### ISSUE TITLE
[P1][dogfood] JS class field declarations (state, private #fields, arrow-function methods) are entirely dropped from extraction

### ISSUE THREAD

**Issue by aimasteracc** [aimasteracc/tree-sitter-analyzer#746]
**Severity:** P1  |  **Surface:** Cross-language extraction correctness (JavaScript) — class members
_Found by the parallel multi-agent dogfood audit on develop @ 471ac5f5; adversarially re-reproduced and dup-checked by an independent verifier agent._

### Observed
All five class members of ModernComponent are reported MISSING: class field `state` (line 33), private fields `#privateCounter` (40) and `#secretKey` (41), and public arrow-function methods `handleClick` (107) and `handleSubmit` (117) appear in neither methods[] nor fields[]. The class is detected (classes:['ModernComponent']) but with 0 of its 5 field/arrow members. Note: private *methods* (#logActivity, #validateData) ARE extracted, so the defect is narrowed to class field declarations (incl. private # fields) and public arrow-function class members.

### Expected
`state`, `#privateCounter`, `#secretKey` should appear as class fields and `handleClick`/`handleSubmit` as class methods (arrow class members). Dropping private fields and arrow-method handlers silently understates a JS/React class's public surface.

### Repro
```bash
cd /Users/aisheng.yu/git-private/tsa-dogfood && uv run python -m tree_sitter_analyzer examples/ModernJavaScript.js --advanced --table json | python3 -c "import json,sys; d=json.load(sys.stdin); mn=[m['name'] for m in d['methods']]; fn=[f['name'] for f in d['fields']]; print({h:('methods' if h in mn else 'fields' if h in fn else 'MISSING') for h in ['state','#privateCounter','#secretKey','handleClick','handleSubmit']})"
# => {'state': 'MISSING', '#privateCounter': 'MISSING', '#secretKey': 'MISSING', 'handleClick': 'MISSING', 'handleSubmit': 'MISSING'}
```

### Verifier notes
Re-ran the exact repro command in /Users/aisheng.yu/git-private/tsa-dogfood on develop. Output: {'state': 'MISSING', '#privateCounter': 'MISSING', '#secretKey': 'MISSING', 'handleClick': 'MISSING', 'handleSubmit': 'MISSING'} — all five members of ModernComponent are absent from both methods[] and fields[]. Confirmed via sed that the source genuinely contains these declarations: `state = {...}` (line 33), `#privateCounter = 0` (40), `#secretKey = '...'` (41), `handleClick = (event) => {...}` (107), `handleSubmit = async (formData) => {...}` (117). So they are not phantom — they exist in source and are dropped by extraction. The class IS detected (classes:['ModernComponent']) but reports 0 of its 5 field/arrow members. This is not a flag/user error: --advanced --table json is the documented advanced-analysis path, and the JSON shape (methods[]/fields[]) is correct; the bug is in the JS extractor's handling of class field declarations and arrow-function class members. Notable nuance refining the finding: private *methods* (#logActivity, #validateData) ARE extracted, so the gap is specifically (a) class field declarations including private # fields, and (b) public arrow-function class members assigned as fields. No open issue covers JS member extraction (searched 'javascript', 'field extraction', 'arrow', 'class field', 'private field', plus full open-issue listing; the only extraction-adjacent open issues #572/#707 concern parse-error surfacing). dup_issue=null. P1 is justified: silently understating a JS/React class's public surface (handlers + state) corrupts the structural picture agents rely on for a very common JS pattern.

**Comment by aimasteracc** [comment]
Fixed in PR #885 (merged 2026-06-15). Class field declarations including private #fields and arrow-function methods now extracted.
